query/src/main/java/org/infinispan/query/impl/IndexInspector.java

```diff
@@ -0,0 +1,81 @@
+package org.infinispan.query.impl;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.hibernate.search.indexes.spi.IndexManager;
+import org.hibernate.search.spi.IndexedTypeIdentifier;
+import org.hibernate.search.spi.SearchIntegrator;
+import org.hibernate.search.spi.impl.PojoIndexedTypeIdentifier;
+import org.infinispan.AdvancedCache;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.query.indexmanager.InfinispanIndexManager;
+
+/**
+ * Provides runtime information about indexing backends.
+ *
+ * @since 11.0
+ */
+@Scope(Scopes.NAMED_CACHE)
+public class IndexInspector {
+
+   private final Map<Class<?>, IndexManager> indexManagerPerClass = new HashMap<>(2);
+   private final Map<String, Class<?>> indexedEntities;
+   private final SearchIntegrator searchFactory;
+
+   public IndexInspector(AdvancedCache<?, ?> cache, SearchIntegrator searchFactory) {
```

Comment:
We don't actually need the Cache here, just its Configuration. Also, when this component is constructed the cache is not really running , just starting, so it would be unsafe to call any methods of the Cache.

Reply:
This seem a leftover, I will get rid of the cache here